cinema/views.py

```diff
@@ -19,26 +26,46 @@
     OrderSerializer,
     OrderListSerializer,
 )
+from cinema.permissions import IsAdminOrIfAuthenticatedReadOnly
+
+
+class ListCreateViewSet(
+    viewsets.GenericViewSet,
+    ListModelMixin,
+    CreateModelMixin
+):
+    pass
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(ListCreateViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
please use DEFAULT_PERMISSION_CLASS and DEFAULT_AUTHENTICATION_CLASSES in settings to avoid duplication in all classes